doc/api/stream.md

```diff
@@ -432,6 +432,18 @@ file.end('world!');
 // Writing more now is not allowed!
 ```
 
+[`stream.end()`][stream-end] will error with (in order of precedence):
+* `ERR_STREAM_WRITE_AFTER_END` if `chunk` is not nully and
+[`stream.end()`][stream-end] has been called.
+* `ERR_STREAM_DESTROYED` if `chunk` is not nully and
+[`stream.destroy()`][stream-destroy] has been called.
+* `ERR_INVALID_ARG_TYPE`  if `chunk` is not nully and
+`chunk` is not a `string` when not in `objectMode`.
+* `ERR_STREAM_ALREADY_FINISHED` if
+[`writable.writableFinished`][].
+* Any error emitted by the instance through `'error'` before
+[`writable.writableFinished`][].
```

Comment:
This last one is not strictly true with the current implementation in master, but I believe that is a bug.

Reply:
Fixed in https://github.com/nodejs/node/pull/29747.

Reply:
Maybe:
```suggestion
[`writable.writableFinished`][] is set to `true`.
```